### ISSUE TITLE
validation loss

### ISSUE THREAD

**Issue by lyu124** [advimman/lama#208]
Hello , i can see from the code that by validation and test step just calculate the evaluation metric of them, but not send them into the trained network to get the same loss metric as the training set. But  i want to calculate the full loss of the validation set as well, is it possible? How can i achieve that? How do we set the optimizer_idx?

Besides that, by training process at each training step we will get a generator loss and a discriminator loss. Does the full loss in code mean the combination of the both losses at each step? 

Thank you so much.

**Comment by windj007** [comment]
> but not send them into the trained network

All train metric (loss parts) are logged also during validation. Except R1, because it need differentiation mode enabled (but validation is under no_grad). Have you tried to run the code?

> the full loss in code

There is no such thing that "full loss" in GANs - it is meaningless because it does not become smaller during training and it is expected. Also, there are separate interleaved steps for generator and discriminator - and on each step only the corresponding loss is calculated.

**Comment by windj007** [comment]
If you point me to the exact lines in the code then I can comment more precisely.

**Comment by Ellohiye** [comment]
> If you point me to the exact lines in the code which make you doubt, then I can comment more precisely.

Hello author, I wonder if you have received my email? Because I sent an attachment, I don't know if it is regarded as a junk file and processed? I've been waiting for your reply and really hope you see my email. thank you very much!

**Comment by lyu124** [comment]
> 

Hello. Thanks a lot. Now i see that.
 Btw, when you mentioned the HRFPL loss, do you mean the combination of feature_matching loss and the resnet_pl loss? Thanks a lot:)

**Comment by windj007** [comment]
HRFPL = high-receptive field perceptual loss = only resnet_pl, feature matching is a separate loss - and it is described in the paper exactly as implemented in the code.
